packages/core/src/icon/icon.element.ts

```diff
@@ -57,7 +59,7 @@ export class CdsIcon extends LitElement {
    * the specified icon cannot be found in the icon registry.
    */
   set shape(val: string) {
-    if (hasStringPropertyChangedAndNotNil(val, this._shape)) {
+    if (hasStringPropertyChangedAndNotNil(val, this._shape) && ClarityIcons.registry[val]) {
```

Comment:
If we are testing for presence in the registry on set, I don't think we need to test for presence on get (line 54)

Reply:
Fixed